rasa/core/actions/action.py

```diff
@@ -625,7 +625,17 @@ async def _utter_responses(
         """Use the responses generated by the action endpoint and utter them."""
         bot_messages = []
         for response in responses:
+            generated_response = None
             generated_response = response.pop("response", None)
+            if not generated_response:
+                generated_response = response.pop("template", None)
+                rasa.shared.utils.io.raise_deprecation_warning(
```

Comment:
should we only raise this warning if `generated_response` from the `template` key `is not None`? If they for some reason send neither of these keys then this warning could be confusing.

Reply:
Great point, fixed